File: server/bcephal-accounting-service/src/main/java/com/moriset/bcephal/accounting/domain/BookingModelScheduledJob.java
/**
 * 
 */
package com.moriset.bcephal.accounting.domain;

import java.util.Date;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.utils.JsonDateTimeDeserializer;
import com.moriset.bcephal.utils.JsonDateTimeSerializer;
/**
 * 
 * @author MORISET-004
 *
 */
public class BookingModelScheduledJob extends BrowserData {

	public String projectName;
	public String projectCode;

	@JsonSerialize(using = JsonDateTimeSerializer.class)
	@JsonDeserialize(using = JsonDateTimeDeserializer.class)
	public Date nextFireTime;

	public boolean currentlyExecuting = false;

	@JsonSerialize(using = JsonDateTimeSerializer.class)
	@JsonDeserialize(using = JsonDateTimeDeserializer.class)
	public Date previousFireTime;

	@JsonSerialize(using = JsonDateTimeSerializer.class)
	@JsonDeserialize(using = JsonDateTimeDeserializer.class)
	public Date currentlyExecutionFireTime;

	public Long runTime;
}

File: client/bcephal-client-model/Accounting/BookingModelScheduledJob.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;
using System;

namespace Bcephal.Kernel.Domain.Accounting
{
    public class BookingModelScheduledJob : BrowserData
    {

        public String projectName { get; set; }

        public String projectCode { get; set; }

        public string nextFireTime { get; set; }

        public bool currentlyExecuting { get; set; }

        public string previousFireTime;

        public string currentlyExecutionFireTime;

        public long? runTime;

        [JsonIgnore]
        public DateTime? nextFireTimeDateTime
        {
            get
            {
                if (string.IsNullOrWhiteSpace(nextFireTime))
                {
                    return null;
                }
                return DateUtils.ParseDateTime(nextFireTime);
            }
        }

        [JsonIgnore]
        public DateTime? previousFireTimeDateTime
        {
            get
            {
                if (string.IsNullOrWhiteSpace(previousFireTime))
                {
                    return null;
                }
                return DateUtils.ParseDateTime(previousFireTime);
            }
        }

        [JsonIgnore]
        public DateTime? currentlyExecutionFireTimeDateTime
        {
            get
            {
                if (string.IsNullOrWhiteSpace(currentlyExecutionFireTime))
                {
                    return null;
                }
                return DateUtils.ParseDateTime(currentlyExecutionFireTime);
            }
        }

    }
}
